pyo3-macros-backend/src/utils.rs

```diff
@@ -337,3 +337,73 @@ pub(crate) fn deprecated_from_py_with(expr_path: &ExprPathWrap) -> Option<TokenS
         }
     })
 }
+
+pub(crate) trait TypeExt {
+    /// Replaces all explicit lifetimes in `self` with elided (`'_`) lifetimes
+    ///
+    /// This is useful if `Self` is used in `const` context, where explicit
+    /// lifetimes are not allowed (yet).
+    fn elide_lifetimes(self) -> Self;
+}
+
+impl TypeExt for syn::Type {
+    fn elide_lifetimes(mut self) -> Self {
```

Comment:
I guess we should add a FIXME somewhere here, and say that we want to remove this again.

Reply:
I'm happy to leave it as-is, we can see how brittle it is in practice (or maybe we end up removing this again when it's unused).